test/e2e/external_entra_create.go

```diff
@@ -0,0 +1,205 @@
+	}
+	if out != nil {
+		if err := json.Unmarshal(b, out); err != nil {
+			return fmt.Errorf("decode response: %w; body=%s", err, string(b))
+		}
+	}
+	return nil
+}
+
+func entraResolveTenantID(ctx context.Context, token string) (string, error) {
+	var org struct {
+		Value []struct {
+			ID string `json:"id"`
+		} `json:"value"`
+	}
+	if err := entraGraphGET(ctx, token, "https://graph.microsoft.com/v1.0/organization?$select=id", &org, http.StatusOK); err != nil {
+		return "", err
+	}
+	if len(org.Value) == 0 || org.Value[0].ID == "" {
+		return "", fmt.Errorf("no organizations returned; ensure `az account set --tenant <TENANT_ID>`")
+	}
+	return org.Value[0].ID, nil
+}
+
+var _ = Describe("ExternalEntra Create", func() {
+	It("creates an Entra app registration and secret, and writes JSON",
+		labels.RequireNothing, labels.Critical, labels.Positive, labels.ExternalEntra, labels.RequireHappyPathInfra,
+		func(ctx context.Context) {
+			token, err := entraGraphToken(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			tenantID, err := entraResolveTenantID(ctx, token)
+			Expect(err).NotTo(HaveOccurred())
+
+			displayName := "e2e-hypershift-oidc"
+			outFile := os.Getenv("ENTRA_E2E_SECRET_PATH")
+			if outFile == "" {
+				outFile = "test/e2e/out/entra_app_secret.json"
+			}
+
+			// 1) Create app
+			createBody := map[string]any{
+				"displayName":    displayName,
+				"signInAudience": "AzureADMyOrg",
+				"web":            map[string]any{"redirectUris": []string{}},
+			}
+			var app entraApplication
+			Expect(entraGraphPOST(ctx, token, "https://graph.microsoft.com/v1.0/applications", createBody, &app, http.StatusCreated)).To(Succeed())
+			Expect(app.ID).NotTo(BeEmpty())
+			Expect(app.AppID).NotTo(BeEmpty())
+
+			// 2) Add secret
+			start := time.Now().UTC().Round(time.Second)
+			end := start.Add(90 * 24 * time.Hour)
+			var addReq entraAddPasswordRequest
+			addReq.PasswordCredential.DisplayName = "e2e-secret"
+			addReq.PasswordCredential.StartDateTime = start
+			addReq.PasswordCredential.EndDateTime = end
+
+			var addResp entraAddPasswordResponse
+			addURL := fmt.Sprintf("https://graph.microsoft.com/v1.0/applications/%s/addPassword", app.ID)
+			Expect(entraGraphPOST(ctx, token, addURL, addReq, &addResp, http.StatusOK)).To(Succeed())
+			Expect(addResp.SecretText).NotTo(BeEmpty())
+
+			// 3) Persist JSON (uses shared entraSecretFile type from entra_shared.go)
+			_ = os.MkdirAll(filepath.Dir(outFile), 0o755)
+			f, err := os.Create(outFile)
+			Expect(err).NotTo(HaveOccurred())
+			defer f.Close()
+
+			Expect(json.NewEncoder(f).Encode(entraSecretFiles{
+				TenantID:        tenantID,
+				AppObjectID:     app.ID,
+				ClientID:        app.AppID,
+				ClientSecret:    addResp.SecretText,
+				DisplayName:     displayName,
+				CreatedAt:       start,
+				SecretExpiresAt: end,
+			})).To(Succeed())
+
+			By("saved Entra app credentials JSON at " + outFile)
+		})
+})
```

Comment:
it looks like all the pieces are here. We'd like to end up with one test that

1. creates the `entra_app_secret.json` registration
2. creates the cluster that uses that client
3. confirms that the cluster actually honors the external OIDC

We won't be running the three independently, we'll be running them in order.